sections/kubecontext.zsh

```diff
@@ -18,24 +18,67 @@ SPACESHIP_KUBECONTEXT_SYMBOL="${SPACESHIP_KUBECONTEXT_SYMBOL="☸️  "}"
 SPACESHIP_KUBECONTEXT_COLOR="${SPACESHIP_KUBECONTEXT_COLOR="cyan"}"
 SPACESHIP_KUBECONTEXT_NAMESPACE_SHOW="${SPACESHIP_KUBECONTEXT_NAMESPACE_SHOW=true}"
 SPACESHIP_KUBECONTEXT_COLOR_GROUPS=(${SPACESHIP_KUBECONTEXT_COLOR_GROUPS=})
+SPACESHIP_KUBECONTEXT_CACHE_DIR="${SPACESHIP_KUBECONTEXT_CACHE_DIR:-"${XDG_CACHE_HOME:-$HOME/.cache}/spaceship-prompt"}"
+SPACESHIP_KUBECONTEXT_CACHE_FILE="${SPACESHIP_KUBECONTEXT_CACHE_DIR}/kubecontext-cache"
 
 # ------------------------------------------------------------------------------
 # Section
 # ------------------------------------------------------------------------------
+kubecontext_file_newer_than() {
+    local left_mtime
+    local right_mtime
+    local left=$1
+    local right=$2
+
+    if stat -c "%s" /dev/null &>/dev/null; then
+        # GNU stat
+        left_mtime=$(stat -L -c %Y ${left})
+        right_mtime=$(stat -L -c %Y ${right})
+    else
+        left_mtime=$(stat -L -f %m ${left})
+        right_mtime=$(stat -L -f %m ${right})
+    fi
+
+    [[ ${left_mtime} -gt ${right_mtime} ]]
+
+}
+kubecontext_cache() {
+    [[ -d ${SPACESHIP_KUBECONTEXT_CACHE_DIR} ]] || mkdir -p ${SPACESHIP_KUBECONTEXT_CACHE_DIR}
+    [[ -r ${SPACESHIP_KUBECONTEXT_CACHE_FILE} ]] || touch -d @0 ${SPACESHIP_KUBECONTEXT_CACHE_FILE}
```

Comment:
This doesn't work on os x.

```
touch: illegal option -- d
usage:
touch [-A [-][[hh]mm]SS] [-acfhm] [-r file] [-t [[CC]YY]MMDDhhmm[.SS]] file ...
stat: /Users/niclas/.cache/spaceship-prompt/kubecontext-cache: stat: No such file or directory
```

The problem here is that os x uses a different `touch`binary. I think it can be fixed with installing coreutils through brew and using `gtouch` instead of `touch`. So a command checker is needed if touch or gtouch should be used.

Reply:
That's embarassing, since I'm on os x but use gnu utils on my path. :)
I will see to how I can fix that without  gnu touch.

I guess `touch -t 197001011200` would also work.

Reply:
OK, I will change it to:
`TZ=UTC touch -t 197001010000 ${SPACESHIP_KUBECONTEXT_CACHE_FILE}`, that seems equivalent.